dependency_manager/src/edm_tool/edm.py

```diff
@@ -1031,15 +1031,29 @@ def main_handler(args):
                   "If this is intendend , please use the --cmake flag to explicitly request this functionality.")
         sys.exit(1)
 
-    workspace = EDM.parse_workspace_directory(working_dir.parent)
+    env_workspace = os.environ.get('EVEREST_EDM_WORKSPACE')
+    workspace_dir = working_dir.parent
+    if env_workspace:
+        workspace_dir = Path(env_workspace).expanduser().resolve()
+        log.info(f'Using workspace path set in EVEREST_EDM_WORKSPACE environment variable: {workspace_dir}')
+    else:
+        log.info(f'Using parent directory as workspace path: {workspace_dir}')
+
+    workspace = EDM.parse_workspace_directory(workspace_dir)
     checkout = EDM.checkout_local_dependencies(workspace, args.workspace, dependencies)
 
     known_branches = ["main", "master"]
     for name, dependency in dependencies.items():
+        shortcut = False
         for checkout_dep in checkout:
             if checkout_dep["name"] == name:
-                continue
+                shortcut = True
+        if shortcut:
+            log.info(f'Dependency "{name}": available locally')
+            continue
+        log.info(f'Dependency "{name}": determining if "{dependency["git_tag"]}" is a tag')
```

Comment:
This this will only check if some rev of the dependency is checked out locally - but will it also check if it corresponds to the correct hash?

Reply:
Or will it only speed up, if the repos are checked out AND mentioned in the local workspace.yaml?

Reply:
Here we actually don't want to check if the hash matches (by design), because it most certainly never will in a locally checked out repo in the "workspace"

Reply:
This might be a bit confusingly named, but this refers to dependencies that are already present in the workspace and that do not need to be gathered via cpm

Reply:
Okay, but then, still all dependencies that are not checked out into the "workspace" will be checked every time cmake configuring is running right?

Reply:
yes. and we can't really do anything else here if we want to make sure not to cache any of these non-workspace dependencies by accident.

- we have to check if they use a real (immutable) "tag". which uses git ls-remoe at the moment, incurring a potentially slow network lookup
  - if it's a tag we shortcut because we won't have caching issues with tags

- however if they use a branch name we unfortunately have to lookup the current HEAD of that branch and point the dependency to this rev. 
  - this is another network access and the operation itself might be quite slow because we have to fetch code from the remote

We do all of this [because of a (design) limitation in CPM](https://github.com/cpm-cmake/CPM.cmake/issues/263)